src/ngraph/runtime/cpu/pass/cpu_workspace_insertion.hpp

```diff
@@ -32,15 +50,18 @@ namespace ngraph
     }
 }
 
-class ngraph::runtime::cpu::pass::CPUWorkspaceInsertion : public ngraph::pass::GraphRewrite
+class ngraph::runtime::cpu::pass::CPUWorkspaceInsertion : public ngraph::pass::FunctionPass
 {
 public:
-    CPUWorkspaceInsertion()
-        : GraphRewrite()
+    CPUWorkspaceInsertion(ngraph::NodeVector& indices_list)
+        : FunctionPass()
+        , m_indices_list(indices_list)
     {
-        construct_max_pool_with_indices();
     }
 
+    virtual bool run_on_function(std::shared_ptr<ngraph::Function> f);
+
 private:
-    void construct_max_pool_with_indices();
+    ngraph::NodeVector& m_indices_list;
```

Comment:
Add comment here about m_indices_list

Reply:
Also make this optional.

Reply:
Then I will have to turn this into a pointer. This seems to be cleaner at an expense of a little bit of clunkiness in API